src/niamoto/core/components/exports/page_generator.py

```
@@ -686,3 +687,379 @@ def get_first_ids(self):
+                depth="../",
+                context={"plots_data": plots_data},
+            )
+
+            return output_path
+
+        except Exception as e:
+            raise TemplateError(
+                "plot_index.html",
+                f"Failed to generate plot index page: {str(e)}",
+                details={"error": str(e)},
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)